### PR TITLE
Run replace template for header keys

### PR DESCRIPTION
**Pull request by joshleeb**
Ref. Substitution not occurring for header key #231 

Run `replace_template` for header keys as well as header values.

### CODE REVIEW

**Comment by joshleeb** [comment]
I've added a test and checked that it will fail without the patch applied in this PR. 

Do you have any suggestions of what to add to the documentation? The current docs already seem to describe the added variable substitution:

> The variables are replaced with real values during test processing.

**Comment by cdent** [comment]
Sorry, for not being explicit, I was thinking in format.rst near the "All of these variables may be used in all of the following fields:" section, adding a short to the request headers description.

But also, when looking at that, I realized you've implemented this for just request_headers, which is sufficient, and worth merging, but if you're feeling up to it, using replace template on the response_headers header name might be useful as well. If you've already spent your personal time budget as much as you want on this, don't worry about it, I'll get to it asap. Having it balanced like that would be good.

Thanks again. I'm glad gabbi's useful for you.

File: gabbi/case.py [hunk 1]
```diff
@@ -487,21 +502,16 @@ def _run_test(self):
         self.url = base_url
         full_url = self._parse_url(base_url)
 
+        # Replace variables in headers with variable values. This includes both
+        # in the header key and the header value.
+        test['request_headers'] = self._replace_headers_template(
+            test['name'], test['request_headers'])
+        test['response_headers'] = self._replace_headers_template(
```
Comment (cdent) [inline gabbi/case.py]: I'll need to look at this more closely in the daylight, but there's a chance the response half of this should be in `HeadersResponseHandler`, but maybe not. I've got insufficient brain at the moment to decide.

**Comment by cdent** [comment]
I'm still not completely decide about whether this shouldn't eventually be processed in the handler, but this is okay as is, adds a good feature, and doesn't preclude other ways of doing it in the future, so: yeah.

**Pull request merged**